### PR TITLE
fix: type normalization

### PR DESCRIPTION
**Pull request by aljazerzen**

### CODE REVIEW

File: prqlc/prql-compiler/src/semantic/resolver/types.rs [hunk 1]
```diff
@@ -500,10 +462,65 @@ pub(crate) fn normalize_type(ty: Ty) -> Ty {
...
+            kind: TyKind::Array(Box::new(normalize_type(*items_ty))),
+            ..ty
+        },
+
         kind => Ty { kind, ..ty },
     }
 }
 
+/// Sinks union into arrays and tuples.
+/// [A] || [B] -> [A || B]
```
Comment (max-sixty) [inline prqlc/prql-compiler/src/semantic/resolver/types.rs:475]: Nitty but should we either use `|` or `||` throughout?

File: prqlc/prql-compiler/src/semantic/resolver/types.rs [hunk 2]
```diff
@@ -500,10 +462,65 @@ pub(crate) fn normalize_type(ty: Ty) -> Ty {
...
+/// {a = A, B} || {c = C, D} -> {a = A, c = C, B || D}
```
Comment (max-sixty) [inline prqlc/prql-compiler/src/semantic/resolver/types.rs:476]: Very clear!

**Review by max-sixty — APPROVED** [review]
Looks great!

File: prqlc/prql-compiler/src/semantic/resolver/types.rs (hunk 1 above)
Comment (aljazerzen) [reply prqlc/prql-compiler/src/semantic/resolver/types.rs:475]: It should be `||`, we've decided on this already.

Comment (max-sixty) [reply prqlc/prql-compiler/src/semantic/resolver/types.rs:475]: Totally fine with me, FYI we do have lots of `|`: https://github.com/PRQL/prql/pull/4013/files#diff-5f3418da7152d1109aefd6a955a037dee9b32049a29f5f750ad52160fc766d40R247

**Pull request merged**